backend/src/v5/middleware/dataConverter/inputs/teamspaces/projects/models/metadata.js

```diff
@@ -56,8 +65,11 @@ const generateSchema = (existingMetadata) => {
 Metadata.validateUpdateCustomMetadata = async (req, res, next) => {
 	try {
 		const { teamspace, container, metadata } = req.params;
-		const containerMetadata = await getMetadataById(teamspace, container, metadata, { metadata: 1 });
-		const schema = generateSchema(containerMetadata.metadata);
+		const existingMetadata = await getMetadataByQuery(teamspace, container, { _id: metadata },
+			{ _id: 0, metadata: 1 });
+		const nonCustomMetadataLookup = existingMetadata.metadata.filter((m) => m.custom !== true)
+			.reduce((a, b) => ({ ...a, [b.key]: 1 }), {});
```

Comment:
should really rename a and b to have better meaning

Reply:
If I use the getCommonElements you mentioned above should I not ditch the lookup ? I can just do a filter and map to create an array of non custom metadata keys, pass it to generateSchema and then find the common elements in there

Reply:
yea, basically your look up becomes array of existing keys

Reply:
either do a flatMap or reduce, don't do .filter().map() though